### PR TITLE
Enqueue All Requests to LocalStorage for Durability

### PR DESCRIPTION
**Pull request by sperand-io**
Opening this up to kick off out-of-band discussion with @tsholmes

### CODE REVIEW

**Comment by sperand-io** [comment]
We'll need a new E2E testing strategy with the queue's processFunction managing sending the requests. Plan is to add some event-emitting functionality to the queue implementation that we can listen on!

**Comment by sperand-io** [comment]
Alrighty. These changes depend on landing https://github.com/segmentio/localstorage-retry/issues/3 upstream and then this should be good to go!

**Comment by sperand-io** [comment]
maybe we should add event listeners and make some `debug` statements when the messages get processed

**Review by tsholmes — APPROVED** [review]
LGTM

**Comment by f2prateek** [comment]
So we're removing `beacon` completely, and making local storage not an opt in anymore?

File: lib/index.js [hunk 1]
```diff
@@ -243,44 +245,26 @@ Segment.prototype.ampId = function(ctx) {
...
   } else {
-    sendAjax();
-  }
-
-  function sendAjax() {
-    // Beacons are sent as a text/plain POST
-    var headers = { 'Content-Type': 'text/plain' };
     send(url, msg, headers, function(err, res) {
-      self.debug('ajax sent %o, received %o', msg, arguments);
-      if (err) return fn(err);
```
Comment (sperand-io) [inline lib/index.js:293]: this fn is only ever called in tests (which is true of how it worked before as well) — should i refactor?

**Review by sperand-io — COMMENTED** [review]
@tsholmes would love your eyes on this plz!

**Comment by sperand-io** [comment]
*Update*: 

- Removes beacon
- Maintains opt-in requirement for retryQueue (for now)
- adds end to end tests for messages that are played through the queue via the queue's new emitter functionality

**Comment by sperand-io** [comment]
the following email has been sent to users of the `beacon` feature: https://paper.dropbox.com/doc/Queuing-customer-comms-HxmUl9ESUwv92OhQR6FMv

**Comment by sperand-io** [comment]
@tsholmes assuming that note goes out by EOD or tomorrow, are you good with merging and deploying this monday? (7/31)

**Comment by tsholmes** [comment]
LGTM. I'm ok with deploying this today.

**Pull request merged**